components/automate-ui/src/app/pages/notifications/notifications.component.html

```diff
@@ -5,6 +5,21 @@
         <chef-heading>Notifications</chef-heading>
         <chef-subheading>Webhook notifications for Infra Client run failures and InSpec compliance scan failures.</chef-subheading>
       </chef-page-header>
+      <app-create-notification-modal
+        [visible]="createModalVisible"
+        [creating]="creatingNotification"
+        [sending]="sendingNotification"
+        [hookStatus]="hookStatus"
+        [notificationRule]="notificationObj"
+        [createForm]="createNotificationForm"
+        [targetKeysValue]="targetKeys"
+        [alertTypeKeysValue]="alertTypeKeys"
+        (close)="closeCreateModal()"
+        [conflictErrorEvent]="conflictErrorEvent"
+        (createClicked)="createNotification()"
+        (sendTestClicked)="sendTestForNotification()"
+        (targetTypeChanged)="updateTargetType()">
+      </app-create-notification-modal>
```

Comment:
This is a red flag--there are way too many inputs here. If you recall, some time ago I suggested the approach of making modals more self-contained. They should manage their own state rather than having the parent do it, when possible. Please take a look and see if you can reduce the surface area.

Reply:
every input required, because we can also avoid using function calls on HTML so we are using the input variable instead of function.

Reply:
@vinay033 You are not quite understanding my suggestion. Took me a bit to dig up my reference example, but please take a look at https://github.com/chef/automate/pull/2704, where I first introduced the **self-contained modal** pattern.

Specifically, what I am suggesting is to move all the logic relevant to _creating_ a notification into `CreateNotificationModalComponent` rather than have it sit in the notifications list page (`NotificationsComponent`), which need only know about _listing_ notifications. Analogously, in the PR I referenced, you will see the logic relevant to creating a user has been move into `CreateUserModalComponent`, because the consumer of that component, `UserManagementComponent`, needs only to know about listing users, not about creating them.

So following the reference PR, you would add an input that is a signal to the modal to open. You can then get rid of the `[visible]` input above; the component manages that property itself.
https://github.com/chef/automate/blob/22e5d7d8dc4c954a6beb0db2c9d238b2b02b8be1/components/automate-ui/src/app/modules/team/team-add-users/team-add-users.component.ts#L186-L188

Then, you move the logic to do the creation as well, which means you can get rid of the `[creating]` input; that will be managed within the modal. Also, testing the notification is done inside the modal (the <kbd>Test Notification</kbd> button), so move that logic in there and that gets rid of the `[sending]` input. 

In a similar fashion, I believe you can remove most if not all of the other inputs. As to what happens when the modal closes, you should not even need to send back any output from `CreateNotificationModalComponent` to `NotificationsComponent`--the subscriptions on the parent page should automatically load the new notification into the list.

Reply:
sorry for the late reply, I followed the PR which you were suggested and I have updated the code part only I need to change in test cases, as soon as I will update the PR.

Reply:
@msorens I have updated the PR, please check it once.